testsupport/wait/member.go

```diff
@@ -344,8 +345,53 @@ func (a *MemberAwaitility) WaitUntilNSTemplateSetDeleted(name string) error {
 	})
 }
 
+type NamespaceWaitCriterion struct {
+	Match func(*corev1.Namespace) bool
+	Diff  func(*corev1.Namespace) string
+}
+
+// UntilNamespaceIsActive returns a `NamespaceWaitCriterion` which checks that the given
+// Namespace is in `Active` phase
+func UntilNamespaceIsActive() NamespaceWaitCriterion {
+	return NamespaceWaitCriterion{
+		Match: func(actual *corev1.Namespace) bool {
+			return actual.Status.Phase == corev1.NamespaceActive
+		},
+		Diff: func(actual *corev1.Namespace) string {
+			return fmt.Sprintf("expected namespace to be active:\n%s", actual.Status.Phase)
+		},
+	}
+}
+
+// UntilNamespaceIsActive returns a `NamespaceWaitCriterion` which checks that the given
+// Namespace is in `Active` phase
+func UntilHasLastAppliedSpaceRoles(expected []toolchainv1alpha1.NSTemplateSetSpaceRole) NamespaceWaitCriterion {
+	return NamespaceWaitCriterion{
+		Match: func(actual *corev1.Namespace) bool {
+			lastAppliedSpaceRoles, found := actual.Annotations[toolchainv1alpha1.LastAppliedSpaceRolesAnnotationKey]
+			if !found {
+				return false
+			}
+			expectedLastAppliedSpaceRoles, _ := json.Marshal(expected) // assume that encoding always works
+			return string(expectedLastAppliedSpaceRoles) == lastAppliedSpaceRoles
+		},
+		Diff: func(actual *corev1.Namespace) string {
+			return "expected namespace to be match annotation"
+		},
+	}
```

Comment:
wouldn't make this the diff better?
```suggestion
	expectedLastAppliedSpaceRoles, _ := json.Marshal(expected) // assume that encoding always works
	return NamespaceWaitCriterion{
		Match: func(actual *corev1.Namespace) bool {
			lastAppliedSpaceRoles, found := actual.Annotations[toolchainv1alpha1.LastAppliedSpaceRolesAnnotationKey]
			if !found {
				return false
			}

			return string(expectedLastAppliedSpaceRoles) == lastAppliedSpaceRoles
		},
		Diff: func(actual *corev1.Namespace) string {
			return fmt.Sprintf("expected namespace to be match annotation,\nExpected: %s\nActual annotations:%v", expectedLastAppliedSpaceRoles, actual.Annotations)
		},
	}
```